src/chain.rs

```diff
@@ -6,6 +6,14 @@
 // polkadot, that only has `const` and `type`s that are used in the runtime, and we can import
 // that.
 
+use codec::{Decode, Encode};
+use jsonrpsee::{core::client::ClientT, rpc_params};
+use once_cell::sync::OnceCell;
+use pallet_transaction_payment::RuntimeDispatchInfo;
+use sp_core::Bytes;
+
+pub static SHARED_CLIENT: OnceCell<SubxtClient> = OnceCell::new();
```

Comment:
this is needed because the `MinerConfig trait` doesn't take &self, so I had to use a global variable fine for now.